### PR TITLE
Nav Native API update

### PR DESCRIPTION
**Pull request by devotaaabel**
This PR Updates the API so that `OfflineNavigator#retrieveRouteFor` takes in a callback instead of blocking the UI thread. This PR also renames a few things.

This is the current structure of navigation (`MapboxNavigation` is the only public facing API):
* `MapboxNavigation` -> `MapboxNavigator` -> `Navigator`

To stick with this structure, this is what I am proposing for offline navigation (`MapboxOfflineRouter` is the only public facing API:
* `MapboxOfflineRouter` -> `OfflineNavigator` -> `Navigator`.

-> in this case denotes a "has a" relationship. If anyone has any opinions about naming, please weigh in.

### CODE REVIEW

File: app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OfflineRerouteActivity.java [hunk 1]
```diff
@@ -200,8 +200,7 @@ public void userOffRoute(Location location) {
     mapboxMap.addMarker(new MarkerOptions().position(new LatLng(location.getLatitude(), location.getLongitude())));
     Point newOrigin = Point.fromLngLat(location.getLongitude(), location.getLatitude());
     OfflineRoute offlineRoute = obtainOfflineRoute(newOrigin, newDestination);
-    route = offlineRouting.findOfflineRoute(offlineRoute);
```
Comment (Guardiola31337) [inline app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OfflineRerouteActivity.java]: > Private field 'route' is never assigned

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/CallbackAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.os.AsyncTask;
+
+/**
+ * Can be extended to add a callback to an async task implementation.
+ */
+public abstract class CallbackAsyncTask<P, Q, R> extends AsyncTask<P, Q, R> {
+  Callback<R> callback;
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/CallbackAsyncTask.java]: > Access can be private

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxOfflineNavigation.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.navigator.Navigator;
+
+public class MapboxOfflineNavigation {
+  private OfflineNavigator offlineNavigator;
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxOfflineNavigation.java]: Have we considered making this `final`?

**Comment by devotaaabel** [comment]
@Guardiola31337 did you want to re-review before I merge?

**Pull request merged**